server/mailet/mailetcontainer-camel/src/main/java/org/apache/james/mailetcontainer/impl/JamesMailetContext.java

```diff
@@ -71,7 +74,7 @@
     /**
      * A hash table of server attributes These are the MailetContext attributes
      */
-    private final Map<String, Object> attributes = new ConcurrentHashMap<>();
+    private final Map<AttributeName, Attribute> attributes = new ConcurrentHashMap<>();
```

Comment:
Stupid question: how do you ensure the AttributeName serving as a key matches the AttributeName of the Attribute value?

Reply:
Thanks to the other methods:

```
    public Object getAttribute(String name) {
        return attributes.get(AttributeName.of(name));
    }

    public void setAttribute(Attribute attribute) {
        attributes.put(attribute.getName(), attribute);
    }
```